hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/Server.java

```diff
@@ -1351,6 +1357,33 @@ private class ResponseParams {
 
     @Override
     public String toString() {
+      boolean isCallerContextEnabled = conf.getBoolean(
+          HADOOP_CALLER_CONTEXT_ENABLED_KEY,
+          HADOOP_CALLER_CONTEXT_ENABLED_DEFAULT);
+      CallerContext context = getCallerContext();
+      if (isCallerContextEnabled && context != null && context.isContextValid()) {
+        String cc = context.getContext();
+        String clientIp = "";
+        String ipKey = CallerContext.CLIENT_IP_STR +
+                CallerContext.Builder.KEY_VALUE_SEPARATOR;
+        int posn = cc.indexOf(ipKey);
+        if (posn != -1) {
+          posn += ipKey.length();
+          int end = cc.indexOf(",", posn);
+          clientIp = (end == -1 ? cc.substring(posn) : cc.substring(posn, end));
```

Comment:
This logic is same as `NameNode#getClientMachine()`, can you move it to the common module and can used by here and `NameNode`.

Reply:
Thanks for the heads up, I'll extract it separately and make it more generic.